Implement user sign up #47

HlibSokol opened the pull request:
During previous tasks the registration logic was already implemented by my colleague @IrinaKrivoruchko, my actual task was to check if it works correctly and to close task on dashboard. I created test for registration and it ran successfully.

![image](https://github.com/user-attachments/assets/0ba5bc3f-aeef-40f0-a386-fb101422078b)


The actual registration (tested in Postman) also runs successfully

![image](https://github.com/user-attachments/assets/3e78ad86-5174-43a3-b640-eb13ba7aaa2b)

Files that can be checked for registration logic:
`users/serializers.py`
`users/test.py`
`users/urls.py`
`users/views.py`

VovaD199 commented on forum/users/tests.py:
To avoid duplicate users:
def test_register_existing_email_should_fail(self):
    self.client.post(self.url, self.payload, format="json")
    response = self.client.post(self.url, self.payload, format="json")
    self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)

VovaD199 commented on forum/users/tests.py:
A test that checks for the absence of, for example, a password:
def test_registration_without_password_should_fail(self):
    data = self.payload.copy()
    data.pop("password")
    response = self.client.post(self.url, data, format="json")
    self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)

HlibSokol commented:
- added 2 tests: to avoid the duplicating users and to fail registration without password
- added comma in `SIMPLE_JWT = {... 'AUTH_HEADER_TYPES': ('Bearer',), ...}` after 'Bearer' so now (...) is tuple, not str
